docs/source/usage/performance-tuning.rst

```diff
@@ -0,0 +1,194 @@
+=======================
+Performance Tuning
+=======================
+
+Data Engineering
+################
+
+It is important for customers to have Data Engineering (DE) tables tuned for performance. Mosaic has been designed to operate directly on standard interchange formats which may appear in various DataFrame columns; currently, WKT, WKB, and GeoJSON are supported, with WKB offering the best properties for storing and querying. Further, while capabilities are always advancing, Mosaic does not expose geometry types as defined in `OGC Simple Features for SQL <https://www.ogc.org/standard/sfs/>`__, making investments in DE all the more beneficial. Additionally, for best tuning, pre-standardize all your geospatial data into the same SRID; for most applications, this should be 4326, see `CRS Docs <https://databrickslabs.github.io/mosaic/usage/grid-indexes-bng.html#coordinate-reference-system>`__. *As of Mosaic 0.3 series, customers need to prepare tables for performant queries, though we are exploring options to improve this experience going forward.*
+
+Polygons & Lines
+****************  
+The best pattern for DE looks something like the following (column names are notional):
+
+1. Call Mosaic's `grid_tessellateexplode <https://databrickslabs.github.io/mosaic/api/spatial-indexing.html#grid-tessellateexplode>`__ on your geometries at a given resolution; in addition to exploding to per row, this adds an ``index`` column, a struct with the following fields: ``is_core``, ``index_id``, ``wkb``; you essentially now have the result of H3 covers operation on the geometries and knowledge of which index space is fully contained within the geometry and WKB chips for the boundaries as well as core index; this is the fundamental call for performance
+2. Call something like ``select index.*, * except(index)`` to pull the struct fields into top level columns; also, adding them first to ensure they are part of the table statistics; you can be more thoughtful based on your data, e.g. don't need WKB in the statistics
```

Comment:
will `select * except(index), index.*` work as well? not sure if `except` has limitations in where it can be called. This will not reorder the original schema

Reply:
Just realised you want them to be first columns, ignore my comment. Though still if you have timestamps you may want them to be close to index columns and there are no real benefits on WKB stats

Reply:
Yessir, this is for DE focus and performance practices to not end up with index columns outside of statistics for wide tables.